### DISCUSSION TITLE
GCode Sender

### DISCUSSION THREAD

**Discussion by duebelpapst** [kachurovskiy/nanoels#227] [General]
Hello Maxim, or whoever answers

I have a problem with saving parts on the controller with the GCode Sender.
I tried it with the instructions you have in the description under GCode but without success., see below.

“To save a program X0\n\Z10 under a name Part1, send “Part1\nX0\n\Z10”. To remove it, send “Part1”. To remove all saved programs, send “”.”

Somehow I don't understand the syntax.

Suppose I want to save the following as Part1:

G1 X10 Z0
Z10
X12 Z12
Z15
X15

What do I have to enter in the Gcode Sender?

Regards
Stefan

**Comment by kachurovskiy** [comment]
Hi Stefan,

See the implementation in https://github.com/kachurovskiy/nanoels/blob/main/h4/h4.ino#L1439-L1478

Double quotes mark start and end of the "saved program" section. If you send `""` - just two double quotes - that means you want to delete all programs.

To execute the GCode you send it without quotes.

To save GCode you send it in quotes with name being the first line:

```
"Part1
G1 X10 Z0
Z10
X12 Z12
Z15
X15"
```

**Comment by duebelpapst** [comment]
Thanks, if you know how, it's easy
